tests/manage/pv_services/test_delete_project_with_pvc_and_pods.py

```diff
@@ -0,0 +1,165 @@
+        cephfs_sc = storageclass_factory(interface=constants.CEPHFILESYSTEM)
+        log.info("Creating RBD Storage class")
+        rbd_sc = storageclass_factory(interface=constants.CEPHBLOCKPOOL)
+
+        log.info("Creating Project 1")
+        project_1 = project_factory()
+
+        log.info("Creating {} CephFS PVCs".format(self.cephfs_pvcs_num))
+        # Generate a given number of CephFS PVCs, randomly assigned RWO or RWX
+        cephfs_pvcs = [pvc_factory(interface=constants.CEPHFILESYSTEM,
+                                   project=project_1,
+                                   storageclass=cephfs_sc,
+                                   access_mode=random.choice(
+                                       [constants.ACCESS_MODE_RWO,
+                                        constants.ACCESS_MODE_RWX]))
+                       for i in range(0, self.cephfs_pvcs_num)]
+        log.info("Creating {} RBD PVCs".format(self.rbd_pvcs_num))
+        rbd_pvcs = multi_pvc_factory(interface=constants.CEPHBLOCKPOOL,
+                                     project=project_1,
+                                     storageclass=rbd_sc,
+                                     access_mode=constants.ACCESS_MODE_RWO,
+                                     num_of_pvc=self.rbd_pvcs_num)
+
+        # Delete the entire Project 1 (along with all of its PVCs)
+        pvs = helpers.get_all_pvs()
+        space_used_before_deletion = check_ceph_used_space()
+        project_deletion_start_time = time.time()
+        #ocp.switch_to_default_rook_cluster_project() #TODO: Is this needed?
+        log.info("Deleting Project 1")
+        project_1.delete(resource_name=project_1.namespace)
+        log.info("Waiting for Project 1 deletion success...")
+        project_1.wait_for_delete(resource_name=project_1.namespace)
+        log.info("Project 1 deleted successfully.")
+        project_deletion_time = time.time() - project_deletion_start_time
+        log.info(
+            "Project 1 deletion time: {} seconds".format(project_deletion_time))
+        # PVs delete might still be running, check until all PVs are deleted
+        # Check time between first PV delete and last PV delete
+        time_between_pvcs_deletion_success = 0
+        # Also check delete time per each PV
+        individual_pv_delete_times = []
+        for pv in pvs:
+            pv_start_time = time.time()
+            pv_name = pv.backed_pv
+            #pv.delete()
+            pv.ocp.wait_for_delete(pv_name)
+            helpers.validate_pv_delete(pv_name)
+            pv_delete_time = time.time() - pv_start_time
+            individual_pv_delete_times.append({pv_name: pv_delete_time})
+            time_between_pvcs_deletion_success += pv_delete_time
+        log.info("{} seconds between first and last PVC deletion".format(
+            time_between_pvcs_deletion_success))
+        # Verify space has been reclaimed
+        space_used_after_deletion = check_ceph_used_space()
```

Comment:
If we delete lazily (and we will), this is not going to work well.

Reply:
Does it mean step 3c in this test case ("Check space is reclaimed") is redundant?